### ISSUE TITLE
Explore options for the positioning service

### ISSUE THREAD

**Issue by pkozlowski-opensource** [ng-bootstrap/core#197]
Candidates are:
* [tether](https://github.com/HubSpot/tether) (used by Bootstrap 4)
* a hand-crafted solution
* code "borrowed" / shared with https://github.com/angular/material2

To take a final decision we should:
* [ ] - explore tether (payload size, dependencies, ease of integration with, state of the project)
* [ ] - explore code in https://github.com/angular/material2 (completeness, API, ease of reuse)
* [ ] - explore feasibility of porting what we've got in angular-ui/bootstrap (cost, known issues, size)

In any case we need to have an interface / API that will abstract whatever we commit to and this API needs to be designed as well.

**Comment by Foxandxss** [comment]
Tether please :P

**Comment by pkozlowski-opensource** [comment]
> Tether please

Sure, here are the tasks before it can happen: 
* explore tether (payload size, dependencies, ease of integration with, state of the project)
* we need to have an interface / API that will abstract whatever we commit to and this API needs to be designed as well.

Any takers?

**Comment by Foxandxss** [comment]
Let's ping @RobJacobs. He knows this kind of stuff.

**Comment by pkozlowski-opensource** [comment]
Update on tether payload => ~7.4 kB (min + gzip). You need CSS on top so it would probably amount to sth ~8kB max (min + gzip)

**Comment by RobJacobs** [comment]
I like the idea of leveraging Angular Material and they are talking about having a component SDK which should make that pretty straight forward.  It looks like they have the position stuff stubbed out here:

https://github.com/angular/material2/tree/master/src/core/overlay/position

**Comment by pkozlowski-opensource** [comment]
@RobJacobs I like this idea as well, generally speaking but in practice:
* I'm not sure how easy / hard would be to extract existing material code
* Provided that it won't be trivial (my assumption for now) - how long would it take to coordinate between 2 teams

I was speaking with Jeremy in MTV and we all are very keen on sharing code, this is not an issue. The issue is: is it feasible in practice? SDK might or might not come and for sure won't come next week. And I would like to finish tooltips / popovers next week :-)

tl;dr; we need practical moves - if anyone could explore / own this topic this would help a lot

**Comment by icfantv** [comment]
We could always just start with the basics (top, bottom, left, right) using portions of our existing $position service (abstracted away, of course so we can plug in other positioning services when ready).

@pkozlowski-opensource you were out, but I'll reiterate (paraphrase) what @robwormald said:  "Sometimes it's better to just get something out there first and _then_ move forward.  It doesn't have to be perfect from day zero."

I know you know this, but just a gentle reminder.  :-)

**Comment by pkozlowski-opensource** [comment]
> "Sometimes it's better to just get something out there first and then move forward. It doesn't have to be perfect from day zero."

I'm not looking for perfection here, I'm looking for something I can use _tomorrow_ and swap if needed without (too much) breaking changes. 

What would be your proposal?

**Comment by icfantv** [comment]
Ok, then I misunderstood your post.
